<a href="https://colab.research.google.com/github/rekha-1414/flyrank-ml-internship/blob/main/work/notebooks/w01_research_question.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# ML-02 — Research Question and Provisional Lane

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## 1. My lane (or freestyle) and why

*Name your lane — or say 'freestyle' and describe your own question. One short paragraph: why this one?*

## My lane

**Lane:** Refresh / Content Opportunity Scoring

I chose this lane because it matches the starter dataset and the example workflow provided in the FlyRank guide. My goal is to identify which content pages should be reviewed first based on observable search and engagement signals. This lane is suitable for a beginner because it starts with a simple baseline and produces a ranked review queue with clear reason codes.

## 2. The question: decision, action, cost of a wrong call

*What decision does your work improve? Who acts on it? What does a wrong recommendation cost?*

## Research Question

**Decision**

Which content pages should be reviewed first for possible content refresh?

**Action**

Create a ranked list of pages using observable search and engagement signals so reviewers can prioritize their work.

**Cost of a wrong call**

If low-priority pages are reviewed first, valuable review time may be wasted and important pages could be delayed.

## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

In [5]:
import pandas as pd

df = pd.read_csv("/content/content_refresh_anonymized.csv")

print("Rows:", len(df))
print("Columns:", len(df.columns))

print("\nTrend direction counts:")
print(df["trend_direction"].value_counts())

Rows: 30000
Columns: 44

Trend direction counts:
trend_direction
down      16262
stable     5962
up         4388
new        2236
flat       1152
Name: count, dtype: int64


The starter dataset contains **30,000 rows** and **44 columns**.

The `trend_direction` column shows five categories:
- down: 16,262
- stable: 5,962
- up: 4,388
- new: 2,236
- flat: 1,152

These numbers show that the dataset contains pages with different performance trends, making it suitable for prioritizing which pages should be reviewed first.

The starter dataset contains **... rows** and **... columns**.

The `trend_direction` column contains categories such as **down**, **stable**, **up**, and others, showing different content performance trends that can be used to rank review candidates.

## 4. Careful words: what I can and can't claim

*Write what your work will be able to say (observed, directional, decision-support) — and what it never will (causal proof, 'predicting Google').*

## Careful Words

This notebook explores observable signals in the starter dataset.

The results may identify pages that appear to be good review candidates, but they do not prove that refreshing a page will improve search performance.

Any recommendations should be reviewed by a human before taking action.

## Self-check

Before you submit, confirm each line honestly:

- [ ] Every section above is filled — markdown thinking AND the code that backs it
- [ ] The notebook runs top to bottom with no errors (Runtime → Run all)
- [ ] No client names, URLs, or private queries anywhere
- [ ] My claims use careful words: observed, measured, directional, decision-support
- [ ] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.